cmd/clusterctl/client/repository/components_client.go

```diff
@@ -47,18 +48,41 @@ func newComponentsClient(provider config.Provider, repository Repository, config
 	}
 }
 
+// GetWithValues returns the components from a repository
+// and replaces variables with their values
+func (f *componentsClient) GetWithValues(version, targetNamespace, watchingNamespace string) (Components, error) {
+	file, err := f.getComponentYaml(version)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewComponentsWithValues(f.provider, version, file, f.configClient, targetNamespace, watchingNamespace)
+}
+
+// Get returns the components from a repository without variable replacement
 func (f *componentsClient) Get(version, targetNamespace, watchingNamespace string) (Components, error) {
+
+	file, err := f.getComponentYaml(version)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewComponents(f.provider, version, file, f.configClient, targetNamespace, watchingNamespace)
```

Comment:
Changing the behavior of an existing method should be clearly advised to the users.
Also, get in clusterctl Get is used for upgrades 
https://github.com/kubernetes-sigs/cluster-api/blob/810240aa628cd8b2dfa655abfbe1596d0b3533f2/cmd/clusterctl/client/cluster/upgrader.go#L328
And this should be replaced with GetWithValues